From 116bd1d0229416ac5f96e87f51654f86e49078f0 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Sat, 22 Sep 2018 02:11:11 +0200
Subject: [PATCH 1/2] shell: Make evinced D-Bus name match application name

D-Bus names are case sensitive, so we need to use org.gnome.Evince.Daemon
not org.gnome.evince.Daemon as its name.

Closes: #990
---
 data/Makefile.am                                              | 2 +-
 ...e.Daemon.service.in => org.gnome.Evince.Daemon.service.in} | 2 +-
 data/org.gnome.Evince.service.in                              | 2 +-
 po/POTFILES.skip                                              | 4 ++--
 shell/Makefile.am                                             | 2 +-
 shell/ev-application.c                                        | 4 ++--
 shell/ev-daemon-gdbus.xml                                     | 2 +-
 shell/ev-daemon.c                                             | 4 ++--
 8 files changed, 11 insertions(+), 11 deletions(-)
 rename data/{org.gnome.evince.Daemon.service.in => org.gnome.Evince.Daemon.service.in} (77%)

diff --git a/data/Makefile.am b/data/Makefile.am
index e254b3de..777ed728 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -34,7 +34,7 @@ desktop_DATA = $(DESKTOP_FILES)
 #
 if ENABLE_DBUS
 servicedir = $(datadir)/dbus-1/services
-service_in_files = org.gnome.evince.Daemon.service.in
+service_in_files = org.gnome.Evince.Daemon.service.in
 service_DATA = $(service_in_files:.service.in=.service)
 
 systemd_user_in_files = org.gnome.Evince.service.in
diff --git a/data/org.gnome.evince.Daemon.service.in b/data/org.gnome.Evince.Daemon.service.in
similarity index 77%
rename from data/org.gnome.evince.Daemon.service.in
rename to data/org.gnome.Evince.Daemon.service.in
index c1cd81c3..8171b099 100644
--- a/data/org.gnome.evince.Daemon.service.in
+++ b/data/org.gnome.Evince.Daemon.service.in
@@ -1,4 +1,4 @@
 [D-BUS Service]
-Name=org.gnome.evince.Daemon
+Name=org.gnome.Evince.Daemon
 Exec=@libexecdir@/evinced
 @if_systemd_service@SystemdService=org.gnome.Evince.service
diff --git a/data/org.gnome.Evince.service.in b/data/org.gnome.Evince.service.in
index 3d19bcc8..b56916e0 100644
--- a/data/org.gnome.Evince.service.in
+++ b/data/org.gnome.Evince.service.in
@@ -4,5 +4,5 @@ Documentation=man:evince(1)
 
 [Service]
 Type=dbus
-BusName=org.gnome.evince.Daemon
+BusName=org.gnome.Evince.Daemon
 ExecStart=@libexecdir@/evinced
diff --git a/po/POTFILES.skip b/po/POTFILES.skip
index 70b71131..388107dc 100644
--- a/po/POTFILES.skip
+++ b/po/POTFILES.skip
@@ -25,7 +25,7 @@ backend/xps/evince-xpsdocument.metainfo.xml.in
 cut-n-paste/unarr/common/conv.c
 data/org.gnome.Evince.desktop.in
 data/org.gnome.Evince-previewer.desktop.in
-data/org.gnome.evince.Daemon.service.in
+data/org.gnome.Evince.Daemon.service.in
 
 sub/backend/comics/evince-comicsdocument.metainfo.xml.in
 sub/backend/djvu/evince-djvudocument.metainfo.xml.in
@@ -36,4 +36,4 @@ sub/backend/tiff/evince-tiffdocument.metainfo.xml.in
 sub/backend/xps/evince-xpsdocument.metainfo.xml.in
 sub/data/org.gnome.Evince.desktop.in
 sub/data/org.gnome.Evince-previewer.desktop.in
-sub/data/org.gnome.evince.Daemon.service.in
+sub/data/org.gnome.Evince.Daemon.service.in
diff --git a/shell/Makefile.am b/shell/Makefile.am
index 092f7780..92c52584 100644
--- a/shell/Makefile.am
+++ b/shell/Makefile.am
@@ -199,7 +199,7 @@ ev-gdbus-generated.c ev-gdbus-generated.h: ev-gdbus.xml Makefile
 
 ev-daemon-gdbus-generated.c ev-daemon-gdbus-generated.h: ev-daemon-gdbus.xml Makefile
 	$(AM_V_GEN) $(GDBUS_CODEGEN) \
-			--interface-prefix=org.gnome.evince \
+			--interface-prefix=org.gnome.Evince \
 			--c-namespace=Ev \
 			--c-generate-object-manager \
 			--generate-c-code ev-daemon-gdbus-generated \
diff --git a/shell/ev-application.c b/shell/ev-application.c
index 42fef8cb..9baa8db9 100644
--- a/shell/ev-application.c
+++ b/shell/ev-application.c
@@ -69,9 +69,9 @@ G_DEFINE_TYPE (EvApplication, ev_application, GTK_TYPE_APPLICATION)
 #define APPLICATION_DBUS_OBJECT_PATH "/org/gnome/evince/Evince"
 #define APPLICATION_DBUS_INTERFACE   "org.gnome.evince.Application"
 
-#define EVINCE_DAEMON_SERVICE        "org.gnome.evince.Daemon"
+#define EVINCE_DAEMON_SERVICE        "org.gnome.Evince.Daemon"
 #define EVINCE_DAEMON_OBJECT_PATH    "/org/gnome/evince/Daemon"
-#define EVINCE_DAEMON_INTERFACE      "org.gnome.evince.Daemon"
+#define EVINCE_DAEMON_INTERFACE      "org.gnome.Evince.Daemon"
 #endif
 
 static void _ev_application_open_uri_at_dest (EvApplication  *application,
diff --git a/shell/ev-daemon-gdbus.xml b/shell/ev-daemon-gdbus.xml
index 6b4f9d15..5cb469dc 100644
--- a/shell/ev-daemon-gdbus.xml
+++ b/shell/ev-daemon-gdbus.xml
@@ -2,7 +2,7 @@
 <!DOCTYPE node PUBLIC "-//freedesktop//DTD D-BUS Introspection 0.1//EN"
                       "http://www.freedesktop.org/software/dbus/introspection.dtd">
 <node>
-  <interface name="org.gnome.evince.Daemon">
+  <interface name="org.gnome.Evince.Daemon">
     <method name="RegisterDocument">
       <arg type="s" name="uri" direction="in"/>
       <arg type="s" name="owner" direction="out"/>
diff --git a/shell/ev-daemon.c b/shell/ev-daemon.c
index 91fdd19a..145b5813 100644
--- a/shell/ev-daemon.c
+++ b/shell/ev-daemon.c
@@ -33,8 +33,8 @@
 
 #include "ev-daemon-gdbus-generated.h"
 
-#define EV_DBUS_DAEMON_NAME             "org.gnome.evince.Daemon"
-#define EV_DBUS_DAEMON_INTERFACE_NAME   "org.gnome.evince.Daemon"
+#define EV_DBUS_DAEMON_NAME             "org.gnome.Evince.Daemon"
+#define EV_DBUS_DAEMON_INTERFACE_NAME   "org.gnome.Evince.Daemon"
 #define EV_DBUS_DAEMON_OBJECT_PATH      "/org/gnome/evince/Daemon"
 
 #define EV_DBUS_WINDOW_INTERFACE_NAME   "org.gnome.evince.Window"
-- 
2.17.1


From 98370393d264c79e99ac2810a95f5825ef8d6398 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Sat, 22 Sep 2018 02:17:23 +0200
Subject: [PATCH 2/2] shell: Make app D-Bus name match application name

Just as for the daemon's name, make it match the application name.
---
 shell/Makefile.am      | 2 +-
 shell/ev-application.c | 2 +-
 shell/ev-daemon.c      | 2 +-
 shell/ev-gdbus.xml     | 4 ++--
 shell/ev-window.c      | 2 +-
 5 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/shell/Makefile.am b/shell/Makefile.am
index 92c52584..a3302fc5 100644
--- a/shell/Makefile.am
+++ b/shell/Makefile.am
@@ -191,7 +191,7 @@ ev-resources.c: $(GRESOURCE_XML_FILE) Makefile $(shell $(GLIB_COMPILE_RESOURCES)
 
 ev-gdbus-generated.c ev-gdbus-generated.h: ev-gdbus.xml Makefile
 	$(AM_V_GEN) $(GDBUS_CODEGEN) \
-			--interface-prefix=org.gnome.evince \
+			--interface-prefix=org.gnome.Evince \
 			--c-namespace=Ev \
 			--c-generate-object-manager \
 			--generate-c-code ev-gdbus-generated \
diff --git a/shell/ev-application.c b/shell/ev-application.c
index 9baa8db9..f58b349b 100644
--- a/shell/ev-application.c
+++ b/shell/ev-application.c
@@ -67,7 +67,7 @@ G_DEFINE_TYPE (EvApplication, ev_application, GTK_TYPE_APPLICATION)
 #ifdef ENABLE_DBUS
 #define APPLICATION_NAME             "org.gnome.Evince"
 #define APPLICATION_DBUS_OBJECT_PATH "/org/gnome/evince/Evince"
-#define APPLICATION_DBUS_INTERFACE   "org.gnome.evince.Application"
+#define APPLICATION_DBUS_INTERFACE   "org.gnome.Evince.Application"
 
 #define EVINCE_DAEMON_SERVICE        "org.gnome.Evince.Daemon"
 #define EVINCE_DAEMON_OBJECT_PATH    "/org/gnome/evince/Daemon"
diff --git a/shell/ev-daemon.c b/shell/ev-daemon.c
index 145b5813..a8e584fe 100644
--- a/shell/ev-daemon.c
+++ b/shell/ev-daemon.c
@@ -37,7 +37,7 @@
 #define EV_DBUS_DAEMON_INTERFACE_NAME   "org.gnome.Evince.Daemon"
 #define EV_DBUS_DAEMON_OBJECT_PATH      "/org/gnome/evince/Daemon"
 
-#define EV_DBUS_WINDOW_INTERFACE_NAME   "org.gnome.evince.Window"
+#define EV_DBUS_WINDOW_INTERFACE_NAME   "org.gnome.Evince.Window"
 
 #define DAEMON_TIMEOUT (30) /* seconds */
 
diff --git a/shell/ev-gdbus.xml b/shell/ev-gdbus.xml
index b6b50d12..dcded61c 100644
--- a/shell/ev-gdbus.xml
+++ b/shell/ev-gdbus.xml
@@ -2,7 +2,7 @@
 <!DOCTYPE node PUBLIC "-//freedesktop//DTD D-BUS Introspection 0.1//EN"
                       "http://www.freedesktop.org/software/dbus/introspection.dtd">
 <node>
-  <interface name='org.gnome.evince.Application'>
+  <interface name='org.gnome.Evince.Application'>
     <annotation name="org.gtk.GDBus.C.Name" value="EvinceApplication" />
     <method name='Reload'>
       <arg type='a{sv}' name='args' direction='in'/>
@@ -12,7 +12,7 @@
       <arg type='ao' name='window_list' direction='out'/>
     </method>
   </interface>
-  <interface name='org.gnome.evince.Window'>
+  <interface name='org.gnome.Evince.Window'>
     <annotation name="org.gtk.GDBus.C.Name" value="EvinceWindow" />
     <method name='SyncView'>
       <arg type='s' name='source_file' direction='in'/>
diff --git a/shell/ev-window.c b/shell/ev-window.c
index 142836ba..177a1233 100644
--- a/shell/ev-window.c
+++ b/shell/ev-window.c
@@ -256,7 +256,7 @@ struct _EvWindowPrivate {
 
 #ifdef ENABLE_DBUS
 #define EV_WINDOW_DBUS_OBJECT_PATH "/org/gnome/evince/Window/%d"
-#define EV_WINDOW_DBUS_INTERFACE   "org.gnome.evince.Window"
+#define EV_WINDOW_DBUS_INTERFACE   "org.gnome.Evince.Window"
 #endif
 
 #define GS_SCHEMA_NAME           "org.gnome.Evince"
-- 
2.17.1

